pkg/controlplane/reconcilers/endpointsadapter_test.go

```diff
@@ -37,59 +38,52 @@ func TestEndpointsAdapterGet(t *testing.T) {
 		endpointSlicesEnabled bool
 		expectedError         error
 		expectedEndpoints     *corev1.Endpoints
-		endpoints             []*corev1.Endpoints
+		initialState          []runtime.Object
```

Comment:
ok, this is runtime.Object because you mix endpoints and endpointslices on the client initialization